com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Breakpoint.java

```diff
@@ -186,13 +193,18 @@ public CompletableFuture<IBreakpoint> install() {
         requests.addAll(newRequests);
 
         if (!newRequests.isEmpty() && !future.isDone()) {
-            this.putProperty("verified", true);
+            updateBreakpointProperties();
             future.complete(this);
         }
 
         return future;
     }
 
+    private void updateBreakpointProperties() {
+        this.putProperty("verified", true);
+        this.putProperty(KEY_FUNCTIONAL, Boolean.valueOf(this.methodSignature != null));
```

Comment:
It should be easier to save the new property into BreakpointRequest directly before L270 when creating BreakpointRequest. Then we can use event.request().getProperty(key) to get it back. 

I think BreakpointRequest.putProperty() exactly is for such kind of scenario. And it will store the property on the client side only and won't be used by JDI.

https://github.com/microsoft/java-debug/blob/2461b037178f16aa0801386d78e3a9400f6fdfbe/com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Breakpoint.java#L265-L272

Reply:
Thanks now I get it. Fixed now.